keanu-python/keanu/network_io.py

```diff
@@ -0,0 +1,74 @@
+from py4j.java_gateway import java_import
+from .context import KeanuContext
+from .base import JavaObjectWrapper
+from .net import BayesNet
+from typing import Dict, Optional
+
+k = KeanuContext()
+
+java_import(k.jvm_view(), "io.improbable.keanu.util.io.ProtobufSaver")
+java_import(k.jvm_view(), "io.improbable.keanu.util.io.DotSaver")
+java_import(k.jvm_view(), "io.improbable.keanu.util.io.JsonSaver")
+java_import(k.jvm_view(), "io.improbable.keanu.util.io.ProtobufLoader")
+java_import(k.jvm_view(), "io.improbable.keanu.util.io.JsonLoader")
+java_import(k.jvm_view(), "java.io.FileOutputStream")
+java_import(k.jvm_view(), "java.io.FileInputStream")
+
+
+def _save_network(saver, filename: str, save_values: bool = False, metadata: Optional[Dict[str, str]] = None):
+    output_stream = k.jvm_view().FileOutputStream(filename)
+    if metadata is None:
+        saver.unwrap().save(output_stream, save_values)
+    else:
+        saver.unwrap().save(output_stream, save_values, k.to_java_map(metadata))
+
+
+class ProtobufSaver(JavaObjectWrapper):
+
+    def __init__(self, net: BayesNet):
+        super(ProtobufSaver, self).__init__(k.jvm_view().ProtobufSaver(net.unwrap()))
+
+    def save(self, filename: str, save_values: bool = False, metadata: Optional[Dict[str, str]] = None):
+        _save_network(self, filename, save_values, metadata)
```

Comment:
can you do this with an abstract base class instead of the static `_save_network` method? It would avoid some code duplication.

Reply:
Ditto for `_load_network`

Reply:
+1

Reply:
Cool - will resolve.